src/Log/LevelFilter.php

```diff
@@ -0,0 +1,89 @@
+ *
+ * OPUS 4 is a complete rewrite of the original OPUS software and was developed
+ * by the Stuttgart University Library, the Library Service Center
+ * Baden-Wuerttemberg, the Cooperative Library Network Berlin-Brandenburg,
+ * the Saarland University and State Library, the Saxon State Library -
+ * Dresden State and University Library, the Bielefeld University Library and
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    opus4-common
+ * @package     Opus\LevelFilter
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace Opus\Log;
+
+/**
+ * Class LevelFilter
+ *
+ * Class for manipulating the filter of a logger like setting the level and disabling it.
+ *
+ * @package Opus\Log
+ */
+class LevelFilter extends \Zend_Log_Filter_Priority
+{
+    private $operator;
+
+    public function __construct($priority, $operator = null)
+    {
+        parent::__construct($priority, $operator);
+
+        $this->operator = $this->_operator;
+    }
+
+    /**
+     * Set the priority of the filter. On null argument, filter is disabled.
+     *
+     * Filter is disabled by setting the priority to lowest value and altering the comparison operator
+     * to greater or equal.
+     *
+     * @param $priority
+     */
+    public function setLevel($priority)
+    {
+        if ($priority === null) {
+            $this->_operator = '>=';
+            $this->_priority = \Zend_Log::EMERG;
+        } elseif ($priority < 0 || ! is_int($priority)) {
+            throw new \InvalidArgumentException('Priority should be of Integer type and cannot be negative');
+        } else {
+            $this->_operator = $this->operator;
+            $this->_priority = $priority;
+        }
+    }
+
+    /**
+     * Returns priority of the filter.
+     *
+     * Returns null if the filter has is disabled.
+     *
+     * @return int|null
+     */
+    public function getLevel()
+    {
+        if ($this->_operator === '>=' && $this->_priority == \Zend_Log::EMERG) {
```

Comment:
I guess this works. It leads to the slightly strange behavior, that if I create a filter with ">=" as operator and set the level to EMERG and then try to get the level I will get NULL. For every other level I will get the level I have set. Now we know why that is and why it isn't a problem, but it is a bit strange and could probably be avoided.

Reply:
Solved.